stream_alert/athena_partition_refresh/main.py

```diff
@@ -0,0 +1,206 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+import json
+import logging
+import os
+
+from datetime import datetime
+
+import backoff
+import boto3
+
+logging.basicConfig()
+LEVEL = os.environ.get('LOGGER_LEVEL', 'INFO')
+LOGGER = logging.getLogger('StreamAlertAthena')
+LOGGER.setLevel(LEVEL.upper())
+
+ATHENA_CLIENT = None
+DATABASE_DEFAULT = 'default'
+
+
+def _load_config():
+    """Load the Athena Lambda configuration file
+
+    Returns:
+        [dict] Configuration settings by file, includes two keys:
+            lambda: All lambda function settings
+            global: StreamAlert global settings
+    """
+    config_files = ('lambda', 'global')
+    config = {}
+    for config_file in config_files:
+        with open('conf/{}.json'.format(config_file)) as config_fh:
+            try:
+                config[config_file] = json.load(config_fh)
+            except ValueError:
+                LOGGER.error('The \'%s\' file could not be loaded into json', config_file)
+                return
+
+    return config
+
+
+def _backoff_handler(details):
+    """Simple logging handler for when polling backoff occurs."""
+    LOGGER.debug('Trying again in %f seconds after %d tries calling %s',
+                 details['wait'],
+                 details['tries'],
+                 details['target'])
+
+
+def _success_handler(details):
+    """Simple logging handler for when polling backoff occurs."""
+    LOGGER.debug('Completed after %d tries calling %s',
+                 details['tries'],
+                 details['target'])
+
+
+def check_query_status(query_execution_id):
+    """Check in on the running query, back off if the job is running or queued
+
+    Returns:
+        [string]: The result of the query, this can be SUCCEEDED, FAILED, or CANCELLED.
+                  Reference http://bit.ly/2uuRtda
+    """
+    @backoff.on_predicate(backoff.fibo,
+                          lambda status: status in ('QUEUED', 'RUNNING'),
+                          max_value=10,
+                          jitter=backoff.full_jitter,
+                          on_backoff=_backoff_handler,
+                          on_success=_success_handler)
+    def _get_query_execution(query_execution_id):
+        return ATHENA_CLIENT.get_query_execution(
+            QueryExecutionId=query_execution_id
+        )['QueryExecution']['Status']['State']
+
+    return _get_query_execution(query_execution_id)
+
+
+def run_athena_query(**kwargs):
+    """Helper function to run Athena queries
+
+    Keyword Args:
+        query [string]: The SQL query to execute
+        database [string]: The database to execute the query against
+        results_bucket [string]: The S3 bucket to store query results
+        results_path [string]: The S3 key to store results
+
+    Returns (one or the other):
```

Comment:
I think the point Austin and I were trying to get at is it'd be nice if a single type is returned vs. a bool or a dict. This could be done via a namedtuple, or by returning the object you reference.